pkg/reconciler/v1alpha1/pipelinerun/pipelinerun_test.go

```diff
@@ -38,174 +39,124 @@ func TestReconcile(t *testing.T) {
 		},
 		Spec: v1alpha1.PipelineRunSpec{
 			PipelineRef: v1alpha1.PipelineRef{
-				Name:       "test-pipeline",
-				APIVersion: "a1",
+				Name: "test-pipeline",
 			},
-		}}, {
+		},
+	}}
+
+	ps := []*v1alpha1.Pipeline{{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "invalid-pipeline",
+			Name:      "test-pipeline",
 			Namespace: "foo",
 		},
-		Spec: v1alpha1.PipelineRunSpec{
-			PipelineRef: v1alpha1.PipelineRef{
-				Name:       "pipeline-not-exist",
-				APIVersion: "a1",
-			},
+		Spec: v1alpha1.PipelineSpec{Tasks: []v1alpha1.PipelineTask{{
+			Name:    "unit-test-1",
+			TaskRef: v1alpha1.TaskRef{Name: "unit-test-task"},
 		}},
+		}},
+	}
+	ts := []*v1alpha1.Task{{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "unit-test-task",
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.TaskSpec{},
+	}}
+	d := testData{
+		prs: prs,
+		ps:  ps,
+		ts:  ts,
+	}
+	c, logs, client := getController(d)
+	err := c.Reconciler.Reconcile(context.Background(), "foo/test-pipeline-run-success")
+	if err != nil {
+		t.Errorf("Did not expect to see error when reconciling valid Pipeline but saw %s", err)
+	}
+	if logs.Len() > 0 {
+		t.Errorf("expected to see no error log. However found errors in logs: %v", logs)
+	}
+	if len(client.Actions()) == 0 {
+		t.Fatalf("Expected client to have been used to create a TaskRun but it wasn't")
+	}
+	actual := client.Actions()[0].(ktesting.CreateAction).GetObject()
+	trueB := true
+	expectedTaskRun := v1alpha1.TaskRun{
```

Comment:
nit: 

```
expectedTaskRun := &v1alpha1.TaskRun{...}
if d := cmp.Diff(actual, expectedTaskRun); d != "" {} // easy to read
```

Reply:
kk! (the change is where we're taking the pointer right? lemme know if i missed something)

Reply:
You got it. `expectedTaskRun` should be a pointer